packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/helpers/validation.ts

```diff
@@ -100,10 +103,30 @@ export class ValidationHelper {
   private getTypeValidator(type: string): SavedObjectsTypeValidator {
     if (!this.typeValidatorMap[type]) {
       const savedObjectType = this.registry.getType(type);
+
+      const stackVersionSchemas =
+        typeof savedObjectType!.schemas === 'function'
+          ? savedObjectType!.schemas()
+          : savedObjectType!.schemas ?? {};
+
+      const modelVersionCreateSchemas =
+        typeof savedObjectType!.modelVersions === 'function'
+          ? savedObjectType!.modelVersions()
+          : savedObjectType!.modelVersions ?? {};
+
+      const combinedSchemas = { ...stackVersionSchemas };
+      Object.entries(modelVersionCreateSchemas).reduce((map, [key, modelVersion]) => {
+        const virtualVersion = modelVersionToVirtualVersion(key);
+        if (modelVersion.schemas?.create) {
+          combinedSchemas[virtualVersion] = modelVersion.schemas!.create!;
+        }
+        return map;
+      }, {});
```

Comment:
nit: using these many `!` makes me a bit anxious. 

I think we can rewrite this in a type-safe way:

```suggestion
      const stackVersionSchemas =
        typeof savedObjectType?.schemas === 'function'
          ? savedObjectType.schemas()
          : savedObjectType?.schemas ?? {};

      const modelVersionCreateSchemas =
        typeof savedObjectType?.modelVersions === 'function'
          ? savedObjectType.modelVersions()
          : savedObjectType?.modelVersions ?? {};

      const combinedSchemas = { ...stackVersionSchemas };
      Object.entries(modelVersionCreateSchemas).reduce((map, [key, modelVersion]) => {
        const virtualVersion = modelVersionToVirtualVersion(key);
        if (modelVersion.schemas?.create) {
          combinedSchemas[virtualVersion] = modelVersion.schemas.create;
        }
        return map;
      }, {});
```